azurelinuxagent/ga/remoteaccess.py

```diff
@@ -0,0 +1,158 @@
+
+from azurelinuxagent.common.event import add_event, WALAEventOperation, elapsed_milliseconds
+from azurelinuxagent.common.exception import ExtensionError, ProtocolError
+from azurelinuxagent.common.future import ustr
+from azurelinuxagent.common.protocol.restapi import ExtHandlerStatus, \
+                                                    ExtensionStatus, \
+                                                    ExtensionSubStatus, \
+                                                    VMStatus, ExtHandler, \
+                                                    get_properties, \
+                                                    set_properties
+from azurelinuxagent.common.protocol.metadata import MetadataProtocol
+from azurelinuxagent.common.utils.cryptutil import CryptUtil
+from azurelinuxagent.common.utils.flexible_version import FlexibleVersion
+from azurelinuxagent.common.utils.processutil import capture_from_process
+from azurelinuxagent.common.protocol import get_protocol_util
+from azurelinuxagent.common.version import AGENT_NAME, CURRENT_VERSION
+from azurelinuxagent.common.osutil import get_osutil
+
+REMOTE_USR_EXPIRATION_FORMAT = "%a, %d %b %Y %H:%M:%S %Z"
+DATE_FORMAT = "%Y-%m-%d"
+TRANSPORT_PRIVATE_CERT = "TransportPrivate.pem"
+REMOTE_ACCESS_ACCOUNT_COMMENT = "JIT Account"
+MAX_TRY_ATTEMPT = 5
+FAILED_ATTEMPT_THROTTLE = 1
+
+def get_remote_access_handler():
+    return RemoteAccessHandler()
+
+class RemoteAccessHandler(object):
+    def __init__(self):
+        self.protocol_util = get_protocol_util()
+        self.protocol = None
+        self.os_util = None
+        self.cryptUtil = CryptUtil(conf.get_openssl_cmd())
+        self.remote_access = None
+        self.incarnation = 0
+
+    def run(self):
+        try:
+            if self.os_util is None:
+                self.os_util = get_osutil()
+            if self.os_util.jit_enabled:
+                self.protocol = self.protocol_util.get_protocol()
+                current_incarnation = self.protocol.get_incarnation()
+                if self.incarnation != current_incarnation:
+                    # something changed. Handle remote access if any.
+                    self.incarnation = current_incarnation
+                    self.protocol.client.update_goal_state(True)
+                    self.protocol.client.update_remote_access_conf(self.protocol.client.goal_state)
+                    self.remote_access = self.protocol.client.remote_access
+                    if self.remote_access is not None:
+                        self.handle_remote_access()
+        except Exception as e:
+            msg = u"Exception processing remote access handler: {0}".format(
+                ustr(e))
+            logger.error(msg)
+            add_event(AGENT_NAME,
+                      version=CURRENT_VERSION,
+                      op=WALAEventOperation.HandleRemoteAccess,
+                      is_success=False,
+                      message=msg)
+            return
```

Comment:
Unnecessary return

Reply:
removed.

---
In reply to: [194026078](https://github.com/Azure/WALinuxAgent/pull/1171#discussion_r194026078) [](ancestors = 194026078)